### PR TITLE
Refactor address and OLC services

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

One more PR to speed up test suite.

Applied similar "client" pattern to `AddressService` and `OLCService`, which uses 3rd party services EasyPost and Google Maps. This time around, the mail address tests are a little too convoluted, so I have only refactored a small portion of them...

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, and performance bottlenecks and written a brief summary in this PR.
- [x] (n/a) If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [x] (n/a) If applicable, my changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How to we demo these changes?

_How does one observe these changes in a deployed system?  Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues
 in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/service/AddressVerificationError.java [hunk 1]
```diff
@@ -1,71 +0,0 @@
-package org.broadinstitute.ddp.service;
-
-import java.util.ArrayList;
-import java.util.List;
-
-/**
- * Creating our class to contain Validation errors. We try to keep the structure of the EasyPost validation error
- */
-public class AddressVerificationError {
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/service/AddressVerificationError.java:9]: Renamed to `ddp.client.EasyPostVerifyError`.

File: pepper-apis/src/test/java/org/broadinstitute/ddp/model/address/OLCPrecisionTest.java [hunk 2]
```diff
@@ -1,94 +0,0 @@
...
-
-import com.google.openlocationcode.OpenLocationCode;
-import org.broadinstitute.ddp.exception.DDPException;
-import org.broadinstitute.ddp.service.OLCService;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.ExpectedException;
-import org.testcontainers.shaded.org.apache.commons.lang.StringUtils;
-
-public class OLCPrecisionTest {
```
Comment (yufengwng) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/model/address/OLCPrecisionTest.java:13]: Tests moved into or was replaced by `OLCServiceTest`.

**Comment by yufengwng** [comment]
please run tests

**Comment by yufengwng** [comment]
please run tests

**Comment by yufengwng** [comment]
please run tests

**Review by raiks2 — APPROVED** [review]
Looks good to me

**Comment by yufengwng** [comment]
please run tests

**Pull request merged**